packages/core-plugin-v1/src/runtime.ts

```diff
@@ -1,297 +1,1555 @@
+      roomId,
+      count: this.getConversationLength(),
       unique: false,
     });
 
-    // use v1 formatMessage
-    const recentMessages = formatMessages({
-      actors: state.actorsData ?? [],
-      messages: recentMessagesData.map((memory: Memory) => {
-        const newMemory = { ...memory };
-        delete newMemory.embedding;
-        return newMemory;
-      }),
+    // Get goals for the room
+    const goalsData = await this.getGoals({
+      agentId: this.agentId,
+      roomId,
+      onlyInProgress: false,
+      count: 10,
     });
 
-    let allAttachments = [];
+    const senderName = actorsData?.find((actor) => actor.id === userId)?.name || 'Unknown';
+    const agentName =
+      actorsData?.find((actor) => actor.id === this.agentId)?.name || this.character.name;
 
-    if (recentMessagesData && Array.isArray(recentMessagesData)) {
-      const lastMessageWithAttachment = recentMessagesData.find(
-        (msg) => msg.content.attachments && msg.content.attachments.length > 0
-      );
+    // Simplified state object - would need to be more comprehensive in production
+    const state: State = {
+      userId,
+      agentId: this.agentId,
+      agentName,
+      senderName,
+      roomId,
+      actorsData,
+      recentMessagesData,
+      goalsData,
+      ...additionalKeys,
+    };
 
-      if (lastMessageWithAttachment) {
-        const lastMessageTime = lastMessageWithAttachment?.createdAt ?? Date.now();
-        const oneHourBeforeLastMessage = lastMessageTime - 60 * 60 * 1000; // 1 hour before last message
+    return state;
+  }
 
-        allAttachments = recentMessagesData
-          .filter((msg) => {
-            const msgTime = msg.createdAt ?? Date.now();
-            return msgTime >= oneHourBeforeLastMessage;
-          })
-          .flatMap((msg) => msg.content.attachments || []);
-      }
+  async updateRecentMessageState(state: State): Promise<State> {
+    console.log(`[Compat Layer] updateRecentMessageState called for room ${state.roomId}`);
+
+    if (!state.roomId) {
+      console.warn('[Compat Layer] updateRecentMessageState: No roomId in state');
+      return state;
     }
 
-    const formattedAttachments = allAttachments
-      .map(
-        (attachment) =>
-          `ID: ${attachment.id}
-Name: ${attachment.title}
-URL: ${attachment.url}
-Type: ${attachment.source}
-Description: ${attachment.description}
-Text: ${attachment.text}
-    `
-      )
-      .join('\n');
+    // Get latest messages
+    const recentMessagesData = await this.messageManager.getMemories({
+      roomId: state.roomId,
+      count: this.getConversationLength(),
+      unique: false,
+    });
 
+    // Return updated state with new messages
     return {
       ...state,
-      recentMessages: addHeader('# Conversation Messages', recentMessages),
       recentMessagesData,
-      attachments: formattedAttachments,
-    } as State;
+    };
+  }
+
+  /**
```

Comment:
why move these into runtime? are there like a generate helper utility for other places?